src/arch/arm/tlb.cc

```diff
@@ -58,8 +58,38 @@ namespace gem5
 
 using namespace ArmISA;
 
+TlbEntry*
+TLB::Table::accessEntry(const KeyType &key)
+{
+    auto entry = findEntry(key);
+
+    if (entry && !key.functional) {
+        accessEntry(entry);
+    }
+
+    return entry;
+}
+
+TlbEntry*
+TLB::Table::findEntry(const KeyType &key) const
+{
+    auto candidates = indexingPolicy->getPossibleEntries(key);
+
+    for (auto candidate : candidates) {
+        auto entry = static_cast<TlbEntry*>(candidate);
+        if (entry->N == key.pagesize && entry->match(key)) {
```

Comment:
Why is the entry->N == key.pagesize not part of the match check?

Reply:
Could you add some of this info to the code? It seems relevant to help avoid bugs in the future.